enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,525 @@
+
+
+The MCN is meant to clarify state reporting for the users and disambiguate these edge cases.
+
+### API Extensions
+
+- Adds the MachineConfigNodes CRD
+
+### Risks and Mitigations
+
+There might be users who do not know to look here. We will mititagte this by reporting in the CO and/or the node status go to
+and look at this object for MCO related node failures or progression.
+
+Adding new MCD node states might have unintended consequences on the current flow which just has "Working" and "Done". Will make sure the
+states I add are either only in between Working and Done OR no matter what we end up back in the "Done" state.
+
+### Drawbacks
+
+The only drawback is increased API usage in the MCO. However, any approach to increased state reporting will increase API calls. This is the most well thought approach to this. 
+
+## Design Details
+
+### Open Questions [optional]
+
+None.
+
+### Test Plan
+
+MCO e2e tests and unit tests will cover this functionality.
+
+### Graduation Criteria
+
+This feature is behind the tech-preview FeatureGate in 4.15. Once it is tested by QE and users it can be GA'd since it should not impact daily usage of a cluster.
+
+## Dev Preview -> Tech Preview
+
+Not applicable. Feature introduced in Tech Preview. 
+
+## Tech Preview -> GA
+
+Bugs that QE finds get fixed and MCN is proven to handle pool status updating properly.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+Between upgrades, this feature should not have a large impact. It will preform as it usually does and it will track node processes in both upgrades and downgrades. If a node is added or removed, the MachineConfigNode object will not report on it until the node is acknowledged by the MCO.
+
+### Version Skew Strategy
+
+Not applicable.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+If the MachineConfigNode cannot be updated, an error is logged but the operator keeps functioning. If an error happens during an update it is represented in the MachineConfigNode clearly.
+
+#### Support Procedures
+
+None.
+
+## Implementation History
+
+As of Openshift 4.15, the feature went under the tech-preview FeatureGate. The basic functionality was implemented and provides users with a way to track node progressions during updates.
+In 4.16, the UncordonedNode column was added to the -o wide output. Many bugs were fixed as well in this version with creation and deletion of nodes.
+
+## Alternatives
+
+Implementation wise, there has been discussion of who should own these objects: the MCD or a new Controller called the MachineStateController. While either approach would work, there are positive and negative impacts to each.
+
+
+## MachineConfigDaemon Approach
```

Comment:
Which approach did we end up with? Which controller is owning/updating the MCN?

Reply:
the Daemon is. I'll add this in.